src/webui/abstractwebapplication.cpp

```diff
@@ -324,17 +268,28 @@ bool AbstractWebApplication::isAuthNeeded()
             || Preferences::instance()->isWebUiLocalAuthEnabled();
 }
 
-void AbstractWebApplication::printFile(const QString& path)
+int AbstractWebApplication::maxAllowedFilesize() const
 {
-    QByteArray data;
-    QString type;
+    return MAX_ALLOWED_FILESIZE;
+}
 
-    if (!readFile(path, data, type)) {
+void AbstractWebApplication::printFile(const QString &path)
+{
+    QFile file {path};
+    if (!file.open(QIODevice::ReadOnly)) {
+        qDebug("File %s was not found!", qUtf8Printable(path));
         status(404, "Not Found");
         return;
     }
+    
+    if (file.size() > maxAllowedFilesize()) {
+        qWarning("%s: exceeded the maximum allowed file size!", qUtf8Printable(path));
+        status(500, "Internal Server Error");
+        print(QObject::tr("Exceeded the maximum allowed file size!"), Http::CONTENT_TYPE_TXT);
```

Comment:
is translation really needed?
also please add `maxAllowedFilesize()` to the message.

Reply:
Apparently, no. But it's translated in all similar places.

Reply:
I just checked them now and most of them are fine except this line (there are 2 instance of it):
`status(400, tr("Incorrect category name"));`
I think you should use `"Bad Request"` (without translation) and add `print(tr("Incorrect category name"));` below.

Reply:
There is also `status(400, "Incorrect torrent hash or name");`

Reply:
Is `Bad Request` correct in these cases?
As said in the doc:
>The HTTP 400 Bad Request response status code indicates that the server could not understand the request due to invalid syntax.

The syntax is correct but the requested resource id isn't. Maybe `404 Not Found`?

Reply:
>Maybe 404 Not Found?

Yes, I was blindly following the old code.